### PR TITLE
python3Packages.sslyze: 3.1.0 -> 4.1.0

### PR DESCRIPTION
**Pull request by veehaitch**
###### Motivation for this change

ZHF: #122042

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veehaitch** [comment]
Result of `nixpkgs-review pr 122175` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python38Packages.nassl</li>
    <li>sslyze (python38Packages.sslyze)</li>
    <li>python38Packages.tls-parser</li>
    <li>python39Packages.nassl</li>
    <li>python39Packages.sslyze</li>
    <li>python39Packages.tls-parser</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122175` at 6a8db92e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>python38Packages.nassl</li>
    <li>sslyze (python38Packages.sslyze)</li>
    <li>python38Packages.tls-parser</li>
    <li>python39Packages.nassl</li>
    <li>python39Packages.sslyze</li>
    <li>python39Packages.tls-parser</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`agpl3` is a deprecated license, please check if project uses `agpl3Plus` or `agpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/sslyze/default.nix:52:5:
```
   |
52 |     license = licenses.agpl3;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`agpl3` is a deprecated license, please check if project uses `agpl3Plus` or `agpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/nassl/default.nix:132:5:
```
    |
132 |     license = licenses.agpl3;
    |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/nassl/default.nix [hunk 1]
```diff
@@ -108,6 +114,10 @@ buildPythonPackage rec {
     invoke package.wheel
   '';
 
+  doCheck = true;
+
+  pythonImportsCheck = [ "nassl" ];
+
   checkInputs = [ pytestCheckHook ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/nassl/default.nix:121]: This doesn't make sense combined with `checkPhase`. Use `pytestFlagsArray`.

File: pkgs/development/python-modules/nassl/default.nix [hunk 2]
```diff
@@ -67,37 +70,40 @@ let
...
+        deps/openssl-OpenSSL_${modernOpenSSLVersion}/
+      ln -s ${opensslStatic.out.dev}/include deps/openssl-OpenSSL_${modernOpenSSLVersion}/include
+      ln -s ${opensslStatic.bin}/bin deps/openssl-OpenSSL_${modernOpenSSLVersion}/apps
 
-    mkdir -p deps/zlib-${zlibVersion}/
-    cp ${zlibStatic.out}/lib/libz.a deps/zlib-${zlibVersion}/
-  '';
+      mkdir -p deps/zlib-${zlibVersion}/
+      cp ${zlibStatic.out}/lib/libz.a deps/zlib-${zlibVersion}/
+    '';
```
Comment (dotlambda) [inline pkgs/development/python-modules/nassl/default.nix:106]: I think we should minimize whitespace changes.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/nassl/default.nix (hunk 1 above)
Comment (veehaitch) [reply pkgs/development/python-modules/nassl/default.nix:119]: Adapted.

File: pkgs/development/python-modules/nassl/default.nix [hunk 3]
```diff
@@ -108,18 +112,22 @@ buildPythonPackage rec {
...
   checkInputs = [ pytestCheckHook ];
 
-  checkPhase = ''
-    # Skip online tests
-    pytest -k 'not Online'
-  '';
+  pytestFlagsArray = [
+    "-k"
+    "'not Online'"
+  ];
```
Comment (jonringer) [inline pkgs/development/python-modules/nassl/default.nix:124]: This is the more succinct way to express this, but essentially does the same thing.
```suggestion
  disabledTests = [
    "Online"
  ];
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

File: pkgs/development/python-modules/nassl/default.nix (hunk 3 above)
Comment (veehaitch) [reply pkgs/development/python-modules/nassl/default.nix]: Thanks for the suggestion! Changed it accordingly 🙂

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 122175 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>6 packages built:</summary>  <ul>    <li>python38Packages.nassl</li>    <li>sslyze (python38Packages.sslyze)</li>    <li>python38Packages.tls-parser</li>    <li>python39Packages.nassl</li>    <li>python39Packages.sslyze</li>    <li>python39Packages.tls-parser</li>  </ul></details>

**Pull request merged**